### PR TITLE
Replace 'ent-search-generic' with 'search-default' pipeline

### PR DESCRIPTION
**Pull request by seanstory**
### part of https://github.com/elastic/search-team/issues/8812

https://github.com/elastic/elasticsearch/pull/107558 added a `search-default-ingestion` pipeline back in 8.15. For 9.0.0, we will switch over to using this as the default, and will no longer create `ent-search-generic-ingestion` on new deployments.

Existing deployments that already have the `ent-search-generic-ingestion` pipeline will continue to have it available to them. At this time, there is no difference between these two other than their name.

### Related PRs
- corresponding Kibana changes: https://github.com/elastic/kibana/pull/204670
- corresponding Connectors changes: https://github.com/elastic/connectors/pull/3049

### CODE REVIEW

**Review by jedrazb — APPROVED** [review]
lgtm

File: .java-version
Comment (joegallo) [inline .java-version:1]: -1 on committing `.java-version` to the repo. Can you fix this please?

Comment (seanstory) [reply .java-version:1]: oof, sorry

Comment (seanstory) [reply .java-version:1]: fix here: https://github.com/elastic/elasticsearch/pull/118971

**Pull request merged**